apps/src/applab/designElements/canvas.jsx

```diff
@@ -47,8 +49,54 @@ var CanvasProperties = React.createClass({
   }
 });
 
+var CanvasEvents = React.createClass({
+  propTypes: {
+    element: React.PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: React.PropTypes.func.isRequired,
+    onInsertEvent: React.PropTypes.func.isRequired
+  },
+
+  getClickEventCode: function() {
+    var id = this.props.element.id;
+    var code =
+      'onEvent("' + id + '", "click", function(event) {\n' +
+      '  console.log("' + id + ' clicked!");\n' +
+      '  setActiveCanvas("' + id + '");\n' +
+      '  circle(event.offsetX, event.offsetY, 10);\n' +
+      '});\n';
```

Comment:
The spec shows moving the turtle to the specified location for screen, so I inferred that doing something similar for canvas might be interesting. (there was no spec for canvas because it was broken when sarah tried to use it)